impl/src/main/java/org/ehcache/loaderwriter/writebehind/AbstractWriteBehindQueue.java

```diff
@@ -398,60 +381,25 @@ private void processBatchedOperations(List<SingleOperation<K, V>> quarantinedIte
-          successes = bulkCacheWritingException.getSuccesses();
-          // remove successful items
-          for (SingleOperation<K, V> singleOperation : batch) {
-            if (successes.contains(singleOperation.getKey()))
-              batch.remove(singleOperation);
-          }
-          if (executionsLeft <= 0) {
-            if (failures != null) {
-              for (Map.Entry<?, Exception> entry : failures.entrySet()) {
-                LOGGER.warn("Exception while processing key '{}' write behind queue", entry.getKey());
-                //TODO : How to handle this properly 
-                if(cacheLoaderWriter instanceof ResilientCacheWriter ) {
-                  ((ResilientCacheWriter<K, V>) cacheLoaderWriter).throwAway((K)entry.getKey(), null, entry.getValue());
-                }
-              }
-            }
-          } else {
-              LOGGER.warn("Exception while processing write behind queue, retrying in {} seconds, {} retries left : {} ", retryAttemptDelaySeconds, executionsLeft, bulkCacheWritingException);
-            try {
-              Thread.sleep(retryAttemptDelaySeconds * MS_IN_SEC);
-            } catch (InterruptedException e1) {
-              Thread.currentThread().interrupt();
-              throw bulkCacheWritingException;
-            }
-          }
-        } catch (Exception e) {
-          if (executionsLeft <= 0) {
-            LOGGER.warn("Exception while bulk processing in write behind queue", e);
-            if(cacheLoaderWriter instanceof ResilientCacheWriter ) {
-              for (SingleOperation<K, V> opr : batch) {
-                ((ResilientCacheWriter<K, V>) cacheLoaderWriter).throwAway(opr.getKey(), opr instanceof WriteOperation ? ((WriteOperation<K, V>)opr).getValue() : null, e);
-              }
-            }
-          } else {
-            LOGGER.warn("Exception while processing write behind queue, retrying in {} seconds, {} retries left : {} ", retryAttemptDelaySeconds, executionsLeft, e);
-            try {
-              Thread.sleep(retryAttemptDelaySeconds * MS_IN_SEC);
-            } catch (InterruptedException e1) {
-              Thread.currentThread().interrupt();
-              throw e;
-            }
+      try {
+        batch.get(0).createBatchOperation(batch).performBatchOperation(cacheLoaderWriter);
+      } catch (BulkCacheWritingException bulkCacheWritingException) {
+        Map<?, Exception> failures = bulkCacheWritingException.getFailures();
+        Set<?> successes = bulkCacheWritingException.getSuccesses();
+        // remove successful items
+        for (SingleOperation<K, V> singleOperation : batch) {
+          if (successes.contains(singleOperation.getKey()))
+            batch.remove(singleOperation);
+        }
+        if (failures != null) {
+          for (Map.Entry<?, Exception> entry : failures.entrySet()) {
+            LOGGER.warn("Exception while processing key '{}' write behind queue", entry.getKey());
```

Comment:
We should delegate to the resilience strategy and remove the set of failed keys from the cache.